src/workerd/api/trace.h

```diff
@@ -19,6 +19,32 @@ class TraceItem;
 class TraceException;
 class TraceLog;
 
+class TailEvent final: public ExtendableEvent {
+public:
+  explicit TailEvent(kj::ArrayPtr<kj::Own<Trace>> tail);
+
+  static jsg::Ref<TailEvent> constructor(kj::String type) = delete;
+  // TODO(soon): constructor?
+
+  // TODO(perf): more efficient to build/return cached array object?  Or iterator?
+  kj::Array<jsg::Ref<TraceItem>> getTail();
+
+  JSG_RESOURCE_TYPE(TailEvent) {
+    JSG_INHERIT(ExtendableEvent);
+
+    JSG_READONLY_INSTANCE_PROPERTY(tail, getTail);
```

Comment:
Is the reason we needed to duplicate the event type to have differently named accessors?

Reply:
Differently named accessors, differently named event, and differently named class. It's unfortunate but there's really no other way to handle it in a backwards compatible way.